packages/gatsby-plugin-gatsby-cloud/src/gatsby-browser.js

```diff
@@ -1,9 +1,40 @@
-import React from "react"
-import Indicator from "./indicator"
+import React, { useEffect, useState } from "react"
+import { createPortal } from "react-dom"
+import Indicator from "./components/Indicator"
 
-export const wrapPageElement = ({ element }) => {
+// export const wrapPageElement = ({ element }) => {
+// if (process.env.GATSBY_PREVIEW_INDICATOR_ENABLED === `true`) {
+//   return <Indicator>{element}</Indicator>
+// } else {
+//   return <>{element}</>
+// }
+// }
+
+function PreviewIndicatorRoot() {
+  const [indicatorRootRef, setIndicatorRootRef] = useState()
+
+  useEffect(() => {
+    const indicatorRoot = document.createElement(`div`)
+    indicatorRoot.id = `gatsby-preview-indicator`
+    setIndicatorRootRef(indicatorRoot)
+    document.body.appendChild(indicatorRoot)
+  }, [])
+
+  if (!indicatorRootRef) {
+    return null
+  } else {
+    return createPortal(<Indicator />, indicatorRootRef)
+  }
```

Comment:
Nit: no need to do if else
```suggestion
  if (!indicatorRootRef) {
    return null
  }
  
  return createPortal(<Indicator />, indicatorRootRef)
```